src/channels.ml

```diff
@@ -77,12 +77,20 @@ let rec sendfile from_fd to_fd len =
     (fun from_fd ->
       with_blocking_fd to_fd
         (fun to_fd ->
+          Lwt_unix.LargeFile.fstat to_fd
+          >>= fun stat ->
+          let maybe_fdatasync () =
+            match stat.Lwt_unix.LargeFile.st_kind with
+            | Unix.S_REG | Unix.S_BLK ->
```

Comment:
So I assume only these two support synchronization

Reply:
Why is `maybe_fdatasync` defined within the sequence of binds? I would have defined it a bit more independently. However, this should have no impact on functionality.

Reply:
It needed access to `stat` and `to_fd`. I moved it to the toplevel now, agreed that its cleaner that way.